### DISCUSSION TITLE
Set a max number of results to fetch

### DISCUSSION THREAD

**Discussion by jgournet** [jertel/elastalert2#925] [Q&A] [answered]
Hi,

Sometimes, one of our services goes bonkers, and can generate a huge amount of errors.
Elastalert then tries to retrieve them all, like this:
```
Queried rule My-App-Error from 2022-08-15 06:39 UTC to 2022-08-15 06:54 UTC: 320000 / 10000 hits (scrolling..)
```

Which in turns crashes elastalert because of lack of memory, AND the email cannot be sent because it's too large anyway.

=> is there a way to set elastalert to not retrieve more than xxxx hits ?
ie: we do want elastalert to retrieve the hits (ie: not use 'count' query), but to make it stop after a certain number.

**Comment by jertel** [comment, marked as answer]
Hello.

From the [docs](https://elastalert2.readthedocs.io/en/latest/elastalert.html#configuration):

![image](https://user-images.githubusercontent.com/16639611/184727585-3c139dbd-44ea-4dba-bff4-25f8b62cf069.png)

**Reply by jgournet** [reply]
Thanks - not sure how I missed it :/
